addons/l10n_it_edi/wizard/account_move_send.py

```diff
@@ -1,12 +1,216 @@
+        for wizard in self:
+            if not wizard.company_id.account_edi_proxy_client_ids.filtered(lambda x: x.proxy_type == 'l10n_it_edi'):
+                wizard.l10n_it_edi_warning = _("You must accept the terms and conditions in the Settings to use the IT EDI.")
+                continue
+            warnings = ""
+            for move in wizard.move_ids:
+                if (move_warnings := move._l10n_it_edi_export_data_check()):
+                    header = move.name + ":" if wizard.mode == 'invoice_multi' else None
+                    warnings += format_errors(header, move_warnings)
+            wizard.l10n_it_edi_warning = warnings
+
+    def _get_proxy_user(self):
+        self.ensure_one()
+        return self.company_id.account_edi_proxy_client_ids.filtered(lambda x: x.proxy_type == 'l10n_it_edi')
+
+    @api.depends('move_ids.journal_id', 'move_ids.invoice_pdf_report_id')
+    def _compute_l10n_it_edi_readonly(self):
+        """ Don't allow the checkbox to be clicked if there already is a PDF,
+            this may lead to inconsistencies between the PDF and the EDI XML
+        """
+        for wizard in self:
+            already_has_pdf = any(wizard.move_ids.mapped("invoice_pdf_report_id"))
+            already_has_xml = any([x._l10n_it_edi_check_processable() for x in wizard.move_ids.mapped("attachment_ids")])
+            wizard.l10n_it_edi_readonly = wizard.l10n_it_edi_warning or already_has_pdf or already_has_xml
+
+    @api.depends('l10n_it_edi_readonly', 'enable_l10n_it_edi')
+    def _compute_l10n_it_edi_checkbox(self):
+        for wizard in self:
+            wizard.l10n_it_edi_checkbox = wizard.enable_l10n_it_edi and not wizard.l10n_it_edi_readonly
+
+    @api.constrains('l10n_it_edi_checkbox')
+    def _check_l10n_it_edi_checkbox(self):
+        for wizard in self:
+            if wizard.l10n_it_edi_checkbox and not wizard.wizard._get_proxy_user():
+                raise UserError(_("You must accept the terms and conditions in the Settings to use the IT EDI."))
+
+    # -------------------------------------------------------------------------
+    # ATTACHMENTS
+    # -------------------------------------------------------------------------
+
+    def _get_mail_attachment_from_doc(self, move):
+        # OVERRIDES 'account'
+        # Don't send EDI XMLs via mail
+        if self.l10n_it_edi_checkbox or move.l10n_it_edi_state:
             return []
-        return super()._get_mail_attachment_from_doc(doc)
+        return super()._get_mail_attachment_from_doc(move)
+
+    # -------------------------------------------------------------------------
+    # BUSINESS ACTIONS
+    # -------------------------------------------------------------------------
+
+    def _hook_invoice_document_before_pdf_report_render(self, invoice, invoice_data):
+        # EXTENDS 'account'
+        super()._hook_invoice_document_before_pdf_report_render(invoice, invoice_data)
+        if self.l10n_it_edi_checkbox:
```

Comment:
loop and add a condition on `_get_default_enable_l10n_it_edi`

Reply:
Loop on what exactly?
![image](https://github.com/odoo/odoo/assets/1665365/23b09562-4f3b-4f00-bdac-0afcaf333e38)